gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -181,6 +197,37 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
         Long.toString(datasetOriginTimestamp), Long.toString(datasetUpstreamTimestamp));
   }
 
+  private List<CommitStep> getCommitSequence(Collection<WorkUnitState> workUnits, Class<?> baseClass)
+      throws IOException {
+    List<CommitStepCopyEntity> steps = Lists.newArrayList();
+    for (WorkUnitState wus : workUnits) {
+      if (baseClass.isAssignableFrom(CopySource.copyEntityClass(wus))) {
+        CommitStepCopyEntity step = (CommitStepCopyEntity) CopySource.deserializeCopyEntity(wus);
+        steps.add(step);
+      }
+    }
+
+    Comparator<CommitStepCopyEntity> commitStepSorter = new Comparator<CommitStepCopyEntity>() {
+      @Override public int compare(CommitStepCopyEntity o1, CommitStepCopyEntity o2) {
+        return Integer.compare(o1.getPriority(), o2.getPriority());
```

Comment:
Steps with lower priority value will execute first, yes. This is the same way that a priority queue would return them (correct me if I'm wrong).